From e4148b210ab029d388f4481d98d558e6c7b275f7 Mon Sep 17 00:00:00 2001
From: BuildTools <unconfigured@null.spigotmc.org>
Date: Thu, 30 Apr 2015 11:57:30 +0200
Subject: [PATCH] mojang-fixxes-4


diff --git a/src/main/java/net/minecraft/server/WerrisChannel.java b/src/main/java/net/minecraft/server/WerrisChannel.java
index 80ca262..ea60285 100644
--- a/src/main/java/net/minecraft/server/WerrisChannel.java
+++ b/src/main/java/net/minecraft/server/WerrisChannel.java
@@ -2,7 +2,7 @@
  * To change this template, choose Tools | Templates
  * and open the template in the editor.
  */
-package de.demonbindestrichcraft.lib.bukkit.wbukkitlib.pluginhooks.register.wrapper.register.myregister.werriscoiniator;
+package net.minecraft.server;
 import io.netty.buffer.ByteBufAllocator;
 import io.netty.channel.ChannelConfig;
 import io.netty.channel.ChannelFuture;
-- 
2.1.4

